lib/sessions.ts

```diff
@@ -85,6 +145,101 @@ export class Sessions {
     });
   }
 
+  /** Parse a JWT and verify the signature, preferring local verification over remote.
+   *
+   * If max_token_age_seconds is set, remote verification will be forced if the JWT was issued at
+   * (based on the "iat" claim) more than that many seconds ago.
+   *
+   * To force remote validation for all tokens, set max_token_age_seconds to zero or use the
+   * authenticate method instead.
+   */
+  async authenticateJwt(
+    jwt: string,
+    options?: {
+      max_token_age_seconds?: number;
+    }
+  ): Promise<{ session: Session; session_jwt: string }> {
+    try {
+      const session = await this.authenticateJwtLocal(jwt, options);
+      return {
+        session,
+        session_jwt: jwt,
+      };
+    } catch (err) {
+      if (err instanceof ClientError && err.code === "jwt_too_old") {
+        // JWT was too old (stale) to verify locally. Check with the Stytch API.
+        return this.authenticate({ session_jwt: jwt });
+      }
+
+      throw err;
+    }
+  }
+
+  /** Parse a JWT and verify the signature locally (without calling /authenticate in the API).
+   *
+   * If maxTokenAge is set, this will return an error if the JWT was issued (based on the "iat"
+   * claim) more than maxTokenAge seconds ago.
+   *
+   * If max_token_age_seconds is explicitly set to zero, all tokens will be considered too old,
+   * even if they are otherwise valid.
+   *
+   * The value for current_date is used to compare timestamp claims ("exp", "nbf", "iat"). It
+   * defaults to the current date (new Date()).
+   */
+  async authenticateJwtLocal(
+    jwt: string,
+    options?: {
+      max_token_age_seconds?: number;
+      current_date?: Date;
+    }
+  ): Promise<Session> {
+    const now = options?.current_date || new Date();
+
+    let payload;
+    try {
+      const token = await jose.jwtVerify(jwt, this.jwksClient, {
+        ...this.jwtOptions,
+        currentDate: now,
```

Comment:
I think we might want to expose `clockTolerance` here too - it won't have much of an effect with long-lived tokens, but certainly with short-lived tokens 30s of clock skew can go a long way

Reply:
Makes sense to me! Implemented in [9b43c6d](https://github.com/stytchauth/stytch-node/pull/116/commits/9b43c6d23a0d4708bd980436fdc69723ccc4e305)